### ISSUE TITLE
Provide basic developer documentation

### ISSUE THREAD

**Issue by Psychlist1972** [microsoft/MIDI#57]

**Comment by Paul-Dempsey** [comment]
Will these docs live in the existing Microsoft learn developer docs repo (perhaps https://github.com/MicrosoftDocs/windows-dev-docs)? 

Or will this be a separate docset in a new repo? (At Microsoft, docs are always maintained in a separate repo).

Who is the Microsoft POC for the written docs?

**Comment by Psychlist1972** [comment]
During preview dev, many other open source developer projects, like C++/WinRT, C#/WinRT, RS/WinRT, all maintained their docs on their source repo. We will do the same here until it comes time to publish in ms learn.

**Comment by Psychlist1972** [comment]
> Will these docs live in the existing Microsoft learn developer docs repo (perhaps https://github.com/MicrosoftDocs/windows-dev-docs)?
> 
> Or will this be a separate docset in a new repo? (At Microsoft, docs are always maintained in a separate repo).
> 
> Who is the Microsoft POC for the written docs?

Related: Are you offering to assist in creating docs?

**Comment by Paul-Dempsey** [comment]
This is an area I might be able to contribute to. I have extensive past experience with Microsoft developer docs, both when I worked at Microsoft (in tech writer, editor, and SDE roles), and as an external contributor.

**Comment by Psychlist1972** [comment]
> This is an area I might be able to contribute to. I have extensive past experience with Microsoft developer docs, both when I worked at Microsoft (in tech writer, editor, and SDE roles), and as an external contributor.

Awesome. Thank you Paul. We'll definitely be able to use your help.

**Comment by Psychlist1972** [comment]
Important note for this from today's research: C++/WinRT has no provision for XML doc creation, so that impacts intellisense and other doc efforts, making anything there entirely manual.

**Comment by Paul-Dempsey** [comment]
Doxygen can produce XML from C++ (and C#, too). There is a tool to generate Markdown from Doxygen XML output: https://github.com/sourcey/moxygen. Might be possible to transform it to Intellisense XML.
(I transitioned from writer to SDE based on documentation tools I wrote, so have experience there ;-).

In the worst case, Doxygen+Moxygen might produce useful first drafts.

**Comment by Psychlist1972** [comment]
The challenge is C++/WinRT. The intellisense and doc info really needs to be generated from the IDL, not the C++. There's a thread here: https://github.com/microsoft/cppwinrt/issues/933

**Comment by Paul-Dempsey** [comment]
Ok, that's a sad picture. I used to work with/in many of the teams involved, and it's all sadly too familiar. Documentation is always a low pri afterthought, down there with localization. Which, weirdly are areas I spent a much of my career in.  There's no way we can fix the C++ language service in Visual Studio from where we sit. So, we''ll have to do it the way we did things in 1985. SOme things never change.
